### PR TITLE
54 facade shedule

### PR DESCRIPTION
**Pull request by Theo-Hafsaoui**
PR to implement the facade *shedule*
#54 
```
.
├── main
│   ├── java
│   │   └── fr
│   │       └── univtln
│   │           └── lhd
│   │               ├── controllers
│   │               ├── exceptions
│   │               ├── facade
│   │               │   └── Schedule.java
│   │               ├── model
│   │               │   └── entities
│   │               │       ├── dao
│   │               │       │   ├── ...
│   │               │       │   ├── slots
│   │               │       │   │   └── ....
│   │               │       │   └── users
│   │               │       │       ├── ....
│   │               │       ├── slots
│   │               │       │   ├── ....
│   │               │       └── users
│   │               │           ├── ....
│   │               └── view
│   │                   ├── ....
```

 - [x] GetSchedule(Student)
 - [x] GetSchedule(Group)
 - [x] Test

### CODE REVIEW

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java [hunk 1]
```diff
@@ -1,7 +1,149 @@
...
+     * @return a List of slot that is in the timerange and with the Groups of the Student
+     */
+    public static List<Slot> getSchedule(Student student, Interval timerange) {
+        List<Slot> slotList = new ArrayList<>();
+        if (student.getId() < 0) {
+            return slotList;
+        }//Might be an edge case
+        SlotDAO dao = SlotDAO.getInstance();
+        for (Group group :
+                student.getStudendGroup()) {
```
Comment (Desmondmies) [inline edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:57]: careful, might conflict with latest changes on student entity

Comment (Theo-Hafsaoui) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:57]: Should not I only use the ID of group

File: edtLHD/src/edittree [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+├── main
+│   ├── java
+│   │   └── fr
+│   │       └── univtln
+│   │           └── lhd
+│   │               ├── controllers
+│   │               ├── exceptions
+│   │               ├── facade
+│   │               │   └── Schedule.java
+│   │               ├── model
```
Comment (anastygnome) [inline edtLHD/src/edittree:11]: Why put this as a file?

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+     */
+    private List<Slot> filterEverySlotOutsideTimerange(List<Slot> slotList, Interval timerange) {
+        List<Slot> slotIntimerange = new ArrayList<>();
+        if (slotList.isEmpty()) {
+            return slotIntimerange;
+        }
+        for (Slot slot :
+                slotList) {
+            if (timerange.encloses(slot.getTimeRange())) {
+                slotIntimerange.add(slot);
```
Comment (anastygnome) [inline edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:124]: I'll accept this for now, but this should really be done in the get query with a where clause at some point

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            }
+        }
+        return slotIntimerange;
+    }
+
+    /**
+     * Take two LocalDate and return an Interval between the two time
+     * to be noted LocalDate doesn't have a time therefor each instant
+     * start at the beginning of the day 00h00
+     * If the two time are on the same day, the all day is return
```
Comment (anastygnome) [inline edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:134]: Typo in comments

File: edtLHD/src/test/java/fr/univtln/lhd/facade/ScheduleTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+class ScheduleTest {
+
+    @Test
+    //Kind of a test
+    void getSchedule() throws SQLException {
+        StudentDAO dao = StudentDAO.getInstance();
+        SlotDAO sdoa = SlotDAO.getInstance();
+        Interval timerange = sdoa.get(3).get().getTimeRange();
+        Student student = dao.get(1).get();
```
Comment (anastygnome) [inline edtLHD/src/test/java/fr/univtln/lhd/facade/ScheduleTest.java:23]: Maybe this should be another PR. We can add tests later on.

**Review by anastygnome — CHANGES_REQUESTED** [review]
Looks good to me, though a few things need to be addressed

File: edtLHD/src/edittree (hunk 2 above)
Comment (Theo-Hafsaoui) [reply edtLHD/src/edittree:11]: What do you mean ?

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java (hunk 3 above)
Comment (Theo-Hafsaoui) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:124]: I agree

File: edtLHD/src/test/java/fr/univtln/lhd/facade/ScheduleTest.java (hunk 5 above)
Comment (Theo-Hafsaoui) [reply edtLHD/src/test/java/fr/univtln/lhd/facade/ScheduleTest.java:23]: that's what I thought too with #66 , do you think we should delete this file ?

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java (hunk 4 above)
Comment (Theo-Hafsaoui) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:134]: where, timerange is not a real word but It seems understandable to me

Comment (anastygnome) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:134]: whole vs all

File: edtLHD/src/test/java/fr/univtln/lhd/facade/ScheduleTest.java (hunk 5 above)
Comment (anastygnome) [reply edtLHD/src/test/java/fr/univtln/lhd/facade/ScheduleTest.java:23]: Yes, I believe it's preferable

File: edtLHD/src/edittree (hunk 2 above)
Comment (anastygnome) [reply edtLHD/src/edittree:11]: This is a file in the commit, is it necessary?

Comment (Theo-Hafsaoui) [reply edtLHD/src/edittree:11]: ah ok didn't saw it before

**Review by anastygnome — APPROVED** [review]

**Review by Desmondmies — APPROVED** [review]

**Pull request merged**